Gemfile

```diff
@@ -9,7 +9,7 @@ gemspec
 SOURCE         = ENV.fetch('SOURCE', :git).to_sym
 REPO_POSTFIX   = SOURCE == :path ? ''                                : '.git'
 DATAMAPPER     = SOURCE == :path ? Pathname(__FILE__).dirname.parent : 'https://github.com/datamapper'
-DM_VERSION     = "~> #{DataMapper::VERSION}"
```

Comment:
I actually think it's better to declare dependencies directly. What I'm stating here is that dm-core depends on the API supported by the 1.2 version of dm- gems or later. In other words, I depend on the version of datamapper above which it is semantically versioned. Datamapper has historically not been semantically versioned, but I think we should going forward.

If we depend on a specific new feature of a newer version of a dm- related gem, then it should be an explicit change to that gem dependency. Otherwise, changing the minimum version should never be necessary until a major bump or a feature change requires it, and then I should hopefully know exactly which gem I'm changing and why I require it (not just changing a broad decoupled min version that affects all gems)